promql/prettier/node.go

```diff
@@ -175,8 +178,10 @@ func (n *nodeInfo) nodeHistory(head parser.Expr, posRange parser.PositionRange,
 			}
 		}
 	case *parser.NumberLiteral, *parser.StringLiteral:
-		nodeMatch = true
-		stack = append(stack, node)
+		if node.PositionRange().Start <= posRange.Start && node.PositionRange().End >= posRange.End {
```

Comment:
This pattern occurs so frequently, you might want to introduce something like

```go
func (*PositionRange) contains(PositionRange) bool
```

in the parser package.

Reply:
Yes, I was about to make that change. Moving progressively towards minute details.

Reply:
Implemented! Hence, resolving.